src/main/java/appeng/core/features/registries/P2PTunnelRegistry.java

```diff
@@ -186,6 +196,16 @@ public void addNewAttunement( @Nonnull final String modId, @Nullable final Tunne
 		this.modIdTunnels.put( modId, type );
 	}
 
+	@Override
+	public void addNewAttunement( @Nonnull final Capability<?> cap, @Nullable final TunnelType type )
+	{
+		if( type == null || cap == null )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2589)